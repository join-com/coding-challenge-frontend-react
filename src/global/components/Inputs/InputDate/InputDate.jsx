import React from 'react';

import PropTypes from 'prop-types';

import { bemCls } from '../../../utils/ClassNameHelpers';

const InputDate = ({
  className, name, label, type, value, placeholder,
  onChange, onBlur,
}) => (
  <div className={`InputDate ${className}`}
  >
    <label
      key={name}
      className={bemCls(`InputDate ${className}`, '__label')}
    >
      {label}
      <input
        type={type}
        className={bemCls(`InputDate ${className}`, '__input')}
        name={name}
        value={value}
        onChange={onChange}
        onBlur={onBlur}
        placeholder={placeholder}
      />
    </label>
  </div>
);

export default InputDate;

InputDate.propTypes = {
  className: PropTypes.string,
  name: PropTypes.string.isRequired,
  label: PropTypes.string,
  type: PropTypes.string,
  value: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
  ]),
  placeholder: PropTypes.string,
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
};

InputDate.defaultProps = {
  className: '',
  label: '',
  type: 'date',
  value: '',
  placeholder: 'MM/DD/YYYY',
  onChange: () => {},
  onBlur: () => {},
};
