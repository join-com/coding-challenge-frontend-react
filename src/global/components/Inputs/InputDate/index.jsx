import React, { Component } from 'react';

import PropTypes from 'prop-types';

import InputDate from './InputDate';

class InputDateWrapper extends Component {
  constructor() {
    super();
    this.handleChange = this.handleChange.bind(this);
    this.handleBlur = this.handleBlur.bind(this);
  }

  handleChange(event) {
    const { name, onChange } = this.props;
    onChange(name, event.target.value);
  }

  handleBlur() {
    const { value, name, onBlur } = this.props;
    onBlur(name, value);
  }

  render() {
    return (
      <InputDate
        {...this.props}
        onChange={this.handleChange}
        onBlur={this.handleBlur}
      />
    );
  }
}

export default InputDateWrapper;

InputDateWrapper.propTypes = {
  name: PropTypes.string,
  value: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
  ]),
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
};

InputDateWrapper.defaultProps = {
  name: '',
  value: '',
  onChange: () => {},
  onBlur: () => {},
};
