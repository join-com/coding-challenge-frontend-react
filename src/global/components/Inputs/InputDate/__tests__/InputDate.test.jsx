import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import InputDate from '../InputDate';

const props = {
  className: 'MyInputClass',
  name: 'myInputName',
  label: 'My Input Label',
  value: 'My Date Value',
  onChange: () => {},
  onBlur: () => {},
};

describe('InputDate', () => {
  it('should render the component correctly with type date', () => {
    const tree = shallow(<InputDate {...props} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
