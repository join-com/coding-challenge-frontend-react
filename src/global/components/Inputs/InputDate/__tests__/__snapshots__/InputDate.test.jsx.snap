// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputDate should render the component correctly with type date 1`] = `
<div
  className="InputDate MyInputClass"
>
  <label
    className="InputDate__label MyInputClass__label"
    key="myInputName"
  >
    My Input Label
    <input
      className="InputDate__input MyInputClass__input"
      name="myInputName"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="MM/DD/YYYY"
      type="date"
      value="My Date Value"
    />
  </label>
</div>
`;
